python/pyspark/executorresourcerequest.py

```diff
@@ -0,0 +1,73 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+class ExecutorResourceRequest(object):
+    """
+    .. note:: Evolving
```

Comment:
Seems it's not an API in Scala side. Seems better to remove this.

Reply:
sorry again you are referring to being public on scala side?
I have the other jira to make those public. I don't see what the big deal is here since its in 3.1.0 anyway. I don't really want to make these private and then go back and change them in a few days to be public.

Reply:
If you want I can make the scala ones public in this pr as well?

Reply:
Thanks, @tgravescs. I am okay with doing separately if that's going to be fixed soon.